docs/designers-developers/developers/tutorials/block-based-themes/README.md

```diff
@@ -0,0 +1,375 @@
+	--wp--preset--line-height--large: 2.5;
+```
+
+**Global styles** are used to set default values for the website and for the blocks.
+
+This example will add the dark grey color as the website background color.
+Add the code inside the globals, after the presets:
+```
+	"styles": {
+		"color": {
+			"background": "var(--wp--preset--color--very-dark-gray)"
+		}
+	}
+```
+
+**Block styles** sets default values for all blocks of a specific type.
+
+This example uses the CSS variables to add text color and line height to the H2 heading block,
+in combination with a custom font size.
+
+When adding styles for the headings block, we need to indicate the heading level, h1 to h6.
+
+Block styles are separate from global styles. Add the code after the globals, but before the closing brace.
+
+```
+"core/heading/h2": {
+	"styles": {
+		"color": {
+			"text": "var( --wp--preset--color--strong-magenta )"
+		},
+		"typography": {
+			"fontSize": "2.5rem",
+			"lineHeight": "var(--wp--preset--line-height--medium)"
+		}
+	}
+},
+```
+
+CSS variables for font sizes are generated using the `editor-font-sizes` theme support or by adding a global preset.
+https://developer.wordpress.org/block-editor/developers/themes/theme-support/#block-font-sizes
+
+If the theme does not add any custom font sizes, variables are created using the default sizes.
+This example adds the default medium font size to the paragraph block.
+
+The font sizes are unit less, which is why we are using calc:
+https://developer.mozilla.org/en-US/docs/Web/CSS/calc
+
+```
+"core/paragraph": {
+	"styles": {
+		"typography": {
+			"fontSize": "calc(1px * var( --wp--preset--font-size--medium ))"
+		}
+	}
+},
+```
+
+Using the CSS variables is optional.
+In this example, the default background color for the group block is changed to white using a color code:
+
+```
+"core/group": {
+	"styles": {
+		"color": {
+			"background": "#ffffff"
+		}
+	}
+}
+```
+
+Below are the presets and styles combined:
+
+```
+{
+	"global": {
+		"presets": {
+			"color": [
+				{
+					"slug": "strong-magenta",
+					"value": "#a156b4"
+				},
+				{
+					"slug": "very-dark-gray",
+					"value": "#444"
+				}
+			],
+			"line-height": [
+				{
+					"slug": "small",
+					"value": "1.3"
+				},
+				{
+					"slug": "medium",
+					"value": "2"
+				},
+				{
+					"slug": "large",
+					"value": "2.5"
+				}
+			]
+		},
+		"styles": {
+			"color": {
+				"background": "var(--wp--preset--color--very-dark-gray)"
+			}
+		}
+	},
+
+	"core/heading/h2": {
+		"styles": {
+			"color": {
+				"text": "var( --wp--preset--color--strong-magenta )"
+			},
+			"typography": {
+				"fontSize": "2.5rem",
+				"lineHeight": "var(--wp--preset--line-height--medium)"
+			}
+		}
+	},
+
+	"core/paragraph": {
+		"styles": {
+			"typography": {
+				"fontSize": "calc(1px * var( --wp--preset--font-size--medium ))"
+			}
+		}
+	},
+
+	"core/group": {
+		"styles": {
+			"color": {
+				"background": "#ffffff"
+			}
+		}
+	}
+}
+```
```

Comment:
To close off this part of the tutorial, I think it would be helpful to point people to the Adding Blocks page.